service/share/discovery/discovery.go

```diff
@@ -0,0 +1,66 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	core "github.com/libp2p/go-libp2p-core/discovery"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+)
+
+const (
+	namespace = "full"
+	interval  = time.Second * 10
+)
+
+// Advertise is a utility function that persistently advertises a service through an Advertiser.
+func Advertise(ctx context.Context, a core.Advertiser) {
+	go func() {
+		for {
+			ttl, err := a.Advertise(ctx, namespace)
+			if err != nil {
+				log.Debugf("Error advertising %s: %s", namespace, err.Error())
+				if ctx.Err() != nil {
+					return
+				}
+
+				select {
+				case <-time.After(interval):
+					continue
+				case <-ctx.Done():
+					return
+				}
+			}
+
+			wait := 7 * ttl / 8
+			select {
+			case <-time.After(wait):
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// FindPeers starts peer discovery every 2 minutes until peer cache will not reach peersLimit.
+// Can be simplified when https://github.com/libp2p/go-libp2p/pull/1379 will be merged.
+func FindPeers(ctx context.Context, d core.Discoverer, n *Notifee) {
+	go func() {
+		t := time.NewTicker(interval * 3)
```

Comment:
In the comment section, we are referring to 2 minutes. However we have a ticker of 30 seconds. 
Can we elaborate the message better?

Reply:
left from previous version. changed to 30 seconds.